### ISSUE TITLE
Deployment to EKS fails with Helm if jobs.image.tag is not specified

### ISSUE THREAD

**Issue by slushysnowman** [clastix/capsule#500]
# Bug description

If jobs.image.tag is not specified in Helm chart values, deployment to EKS fails.

This is because if not specified manually, the tag for the images used for the jobs is constructed from the kube version of the cluster in https://github.com/clastix/capsule/blob/master/charts/capsule/templates/_helpers.tpl:
```
{{- $kubeversion := print "v" .Capabilities.KubeVersion.Major "." .Capabilities.KubeVersion.Minor -}}
{{- printf "%s:%s" .Values.jobs.image.repository $kubeversion -}}
```

On EKS the minor version contains a + as shown below:
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

This results in an image tag of `v1.21+` which is an invalid format
```
        message: 'Failed to apply default image tag "our-private-registry.com/clastix/kubectl:v1.21+":
          couldn''t parse image reference "our-private-registry.com/clastix/kubectl:v1.21+":
          invalid reference format'
```

Workaround is just to hardcode the jobs.image.tag in values.yaml

# How to reproduce

1. Spin up an EKS cluster
2. Use helm to install version 0.1.5
3. post-install and pre-delete jobs will fail due to inability to pull images due to invalid image tag

# Expected behavior

I'd expect the jobs to work without having to manually specify an image tag - maybe any invalid characters can be stripped from the constructed tag? Not sure what the best solution here is to be honest - maybe the best course of action is documentation if a reliable solution can't be found?

# Logs
```
        message: 'Failed to apply default image tag "our-private-registry.com/clastix/kubectl:v1.21+":
          couldn''t parse image reference "our-private-registry.com/clastix/kubectl:v1.21+":
          invalid reference format'
```
# Additional context

- Capsule version: 0.1.1
- Helm Chart version: 0.1.5
- Kubernetes version: 
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-06eac09", GitCommit:"5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97", GitTreeState:"clean", BuildDate:"2021-09-13T14:20:15Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}

**Comment by prometherion** [comment]
The server version reply is really odd, I mean:

```
Minor:"21+"
```

I never saw a Kubernetes API Server returning a non-semantic versioning value.

It seems more a rare case rather than a bug, we can try to document this in our guide on setting up Capsule on AWS EKS: WDYT?

**Comment by slushysnowman** [comment]
Yeah I also find it super weird and Im pretty sure this is a specific EKS thing - we also run AKS and had no problems there at all (which is where I did my testing and then ran into this when I pushed to our staging env on AWS, woops...)

I don't know to be honest, it'd be nice if there was an automatic way of doing it, but it's gonna add more complexity into the the _helpers.tpl which I don't know if is ideal (check if minor version has + suffix and remove if so).

On the flip side I can imagine there are quite a lot of users running on EKS, so maybe it's worth it, so that they don't have to maintain the tag version in values.yaml.

If you want I can see what I can come up with, create a PR for it, and if it's undesirable to add that complexity in then just document instead?

**Comment by prometherion** [comment]
What about using the Helm function `replace`?

Something as:

```
{{- $kubeversion := print "v" .Capabilities.KubeVersion.Major "." (.Capabilities.KubeVersion.Minor | replace "+" "") -}}
```

Not sure this is going to work, would be great if you could try on your own :)

**Comment by slushysnowman** [comment]
Yeah that'd work and I guess it's not too complex - I'll test it and see how it goes and create a PR if it seems OK, then you guys can decide if you want it or not ;)
